code/code/cmd/cmd_deathstroke.cc

```diff
@@ -35,96 +41,115 @@ static int deathstroke(TBeing *caster, TBeing *victim)
-    if (caster->reconcileDamage(victim, dam,sktype) == -1)
+    if (caster->reconcileDamage(victim, dam,sktype) == -1) 
       return DELETE_VICT;
-  } else {
+  }
+  // Unsuccessful skill check
+  else {
     if (victim->getPosition() > POSITION_DEAD) {
       act("$n's attempt at $N's vital area falls far short of hitting.", 
                   FALSE, caster, 0, victim, TO_NOTVICT);
       act("You fail to hit $N's vital area.", FALSE, caster, 0, victim, 
                   TO_CHAR);
       act("$n attempts to hit your vital area, but fails miserably.", 
                   FALSE, caster, 0, victim, TO_VICT);
-      if (caster->reconcileDamage(victim, 0,sktype) == -1)
-        return DELETE_VICT;
     }
 
-    // with great failure, comes great sorrow *grin*
+    if (caster->reconcileDamage(victim, 0,sktype) == -1) 
+      return DELETE_VICT;
+  }
+  // Applying the debuff and, on success, the buff
+  if (wasSuccess) {
+    caster->affectJoin(caster, &aff1, AVG_DUR_YES, AVG_EFF_YES, FALSE);
+    caster->affectJoin(caster, &aff2, AVG_DUR_YES, AVG_EFF_YES, FALSE);
+  }
+  else {
+    caster->affectTo(&aff1, -1);
+  }
+
     // what happens here is that the mob gets a shot at the players vitals
     // ... fair is fair right? 
-    percent = ((10 - (caster->getArmor() / 50)) << 1);
+    percent = ((10 - (caster->getArmor() / 20)) << 1);
     percent += victim->getDexReaction() * 5;
     percent -= caster->getAgiReaction() * 5;
+    percent -= caster->getSpeReaction() * 3;
+
     int bKnown2 = victim->getSkillValue(SKILL_DEATHSTROKE);
+
     if (bKnown2 > 0 &&
         victim->bSuccess(bKnown2 + percent, SKILL_DEATHSTROKE) &&
          (i = victim->specialAttack(caster,SKILL_DEATHSTROKE)) &&
          i != GUARANTEED_FAILURE) {
       /* victim hits attacker vitals while attacker is exposed */
       CF(SKILL_DEATHSTROKE);
       if (victim->getPosition() > POSITION_STUNNED) {
+
+        dam = victim->GetMaxLevel()*2;
+        dam += victim->plotStat(STAT_CURRENT, STAT_STR, 0, 6, 3);
+        dam = ::number(victim->GetMaxLevel()/3, dam);
+        dam = victim->getActualDamage(victim, NULL, dam, SKILL_DEATHSTROKE);
+        dam /= 3;
+
         act("$N exploits $n's vulnerable state with a quick hit to the heart.",
-                FALSE, caster, 0, victim, TO_NOTVICT);
+              FALSE, caster, 0, victim, TO_NOTVICT);
         act("While you are vulnerable, $N strikes you in the center of your torso.",
-                FALSE, caster, 0, victim, TO_CHAR);
+              FALSE, caster, 0, victim, TO_CHAR);
         act("You take advantage of $n's vulnerability for a cheap shot!", 
-               FALSE, caster, 0, victim, TO_VICT);
-
-        dam = 3*victim->GetMaxLevel();
-        dam += 3*victim->plotStat(STAT_CURRENT, STAT_STR, 0, 6, 3);
-        dam = ::number(1, dam);
-        dam = victim->getActualDamage(victim, NULL, dam, SKILL_DEATHSTROKE);
-        dam /= 2;
+             FALSE, caster, 0, victim, TO_VICT);
```

Comment:
If you're still adding up the `percent` value here and passing it to `bSuccess` doesn't that mean you're giving the victim bonuses when they attempt their counter-deathstroke? Or am I reading this wrong?